gpu_support/add_nvidia_gpu_support.sh

```diff
@@ -0,0 +1,203 @@
+fi
+
+EESSI_SILENT=1 source /cvmfs/pilot.eessi-hpc.org/latest/init/bash
+
+##############################################################################################
+# Check that the CUDA driver version is adequate
+# (
+#  needs to be r450 or r470 which are LTS, other production branches are acceptable but not
+#  recommended, below r450 is not compatible [with an exception we will not explore,see
+#  https://docs.nvidia.com/datacenter/tesla/drivers/#cuda-drivers]
+# )
+# only check first number in case of multiple GPUs
+if [[ "${install_wo_gpu}" != "true" ]]; then
+  driver_major_version=$(nvidia-smi --query-gpu=driver_version --format=csv,noheader | tail -n1)
+  driver_major_version="${driver_major_version%%.*}"
+  # Now check driver_version for compatability
+  # Check driver is at least LTS driver R450, see https://docs.nvidia.com/datacenter/tesla/drivers/#cuda-drivers
+  if (( $driver_major_version < 450 )); then
+    echo "Your NVIDIA driver version is too old, please update first.."
+    exit 1
+  fi
+fi
+
+###############################################################################################
+###############################################################################################
+# Install CUDA
+# TODO: Can we do a trimmed install?
+# if modules dir exists, load it for usage within Lmod
+cuda_install_dir="${EESSI_SOFTWARE_PATH/versions/host_injections}"
+mkdir -p ${cuda_install_dir}
+# only install CUDA if specified version is not found
+if [ -d ${cuda_install_dir}/software/CUDA/${install_cuda_version} ]; then
+  echo "CUDA module found! No need to install CUDA again, proceeding with tests"
+else
+  # - as an installation location just use $EESSI_SOFTWARE_PATH but replacing `versions` with `host_injections`
+  #   (CUDA is a binary installation so no need to worry too much about this)
+  # TODO: The install is pretty fat, you need lots of space for download/unpack/install (~3*5GB), need to do a space check before we proceed
+  avail_space=$(df --output=avail ${cuda_install_dir}/ | tail -n 1 | awk '{print $1}')
+  if (( ${avail_space} < 16000000 )); then
+    echo "Need more disk space to install CUDA, exiting now..."
+    exit 1
+  fi
+  # install cuda in host_injections
+  module load EasyBuild
+  # we need the --rebuild option, since the module file is shipped with EESSI
+  tmpdir=$(mktemp -d)
+  eb --rebuild --installpath-modules=${tmpdir} --installpath=${cuda_install_dir}/ CUDA-${install_cuda_version}.eb
+  ret=$?
+  if [ $ret -ne 0 ]; then
+    echo "CUDA installation failed, please check EasyBuild logs..."
+    exit 1
+  fi
+  rm -rf ${tmpdir}
+fi
+
+# install p7zip, this will be used to install the CUDA compat libraries from rpm
+# the rpm and deb files contain the same libraries, so we just stick to the rpm version
+module avail 2>&1 | grep -i p7zip &> /dev/null
+if [[ $? -eq 0 ]]; then
+  echo "p7zip module found! No need to install p7zip again, proceeding with installation of compat libraries"
+else
```

Comment:
Probably best to just error out here, `p7zip` should be part of the software layer (it isn't yet but it will be)

Reply:
You are right to leave this for now, it's required if anyone wants to test this (since `p7zip` is still not part of EESSI)